core/src/main/java/bisq/core/locale/CurrencyUtil.java

```diff
@@ -62,6 +62,10 @@ public static void setup() {
 
     private static String baseCurrencyCode = "BTC";
 
+    // Calls to isFiatCurrency and isCryptoCurrency are very frequent so we use a cache of the results
+    private static final Map<String, Boolean> isFiatCurrencyMap = new HashMap<>();
+    private static final Map<String, Boolean> isCryptoCurrencyMap = new HashMap<>();
```

Comment:
I think these should be concurrent hash maps, to preserve thread safety (unless it's certain that the two methods will only be called from the user thread).

Reply:
Made a commit with the suggestion